frontend/src/scenes/billing/Billing.js

```diff
@@ -0,0 +1,174 @@
+import React, { useEffect, useState } from 'react'
+import { useValues, useActions } from 'kea'
+import { billingLogic } from './billingLogic'
+import { Card, Progress, Row, Col, Button, Popconfirm, Spin } from 'antd'
+import PropTypes from 'prop-types'
+import defaultImg from './../../../public/plan-default.svg'
+
+const Plan = (props) => {
+    const { plan, onUpgrade } = props
+    return (
+        <Card>
+            <img src={plan.image_url || defaultImg} alt="" height={100} width={100} />
+            <h3 style={{ fontSize: 22 }}>{plan.name}</h3>
+            <div>
+                <Popconfirm
+                    title={`Sign up for the ${plan.name} now? You will need a bank card.`}
+                    onConfirm={() => onUpgrade(plan)}
+                    okText="Yes"
+                    cancelText="No"
+                >
+                    <Button data-attr="btn-upgrade-now" data-plan={plan.key}>
+                        Upgrade now
+                    </Button>
+                </Popconfirm>
+            </div>
+        </Card>
+    )
+}
+
+function Billing(props) {
+    const logic = billingLogic()
+    const { plans, billingSubscription, billingSubscriptionLoading } = useValues(logic)
+    const { loadPlans, subscribe } = useActions(logic)
+    const [state, setState] = useState({ percentage: 0 })
+    const { user } = props
+
+    const strokeColor = (percentage) => {
+        let color = '#1890FF'
+        if (percentage === null || percentage === undefined) {
+            /* No event limit set */
+            color = {
+                from: '#1890FF',
+                to: '#52C41A',
+            }
+        }
+
+        if (percentage > 0.65 && percentage < 0.8) {
+            color = '#F7A501'
+        }
+        if (percentage > 0.8) {
+            color = '#F54E00'
+        }
+        return color
+    }
+
+    const handleBillingSubscribe = (plan) => {
+        subscribe(plan.key)
+    }
+
+    useEffect(() => {
+        if (!user.billing?.plan || user.billing?.should_setup_billing) loadPlans()
+        if (!user.billing?.current_usage || !user.billing.plan) return
+        if (!user.billing.plan.allowance) {
+            /* Plan is unlimited */
+            setState({ ...state, percentage: null })
+            return
+        }
+        const percentage =
+            Math.round((user.billing.current_usage.value / user.billing.plan.allowance.value) * 100) / 100
+        setState({ ...state, percentage })
+    }, [user])
+
+    useEffect(() => {
+        if (billingSubscription?.subscription_url) window.location.href = billingSubscription.subscription_url
+    }, [billingSubscription])
```

Comment:
This can go inside some `listener` in `billingLogic`. 

Also, is the `subscription_url` an external link? If so, using `window.location.href` makes sense here. Otherwise if it's an internal link that should be redirected to without a full browser redirect, you can use this:

```js
import { router } from 'kea-router'

router.actions.push(billingSubscription.subscription_url)
```

Reply:
It is an internal route, but a Django route, not on the React frontend, so keeping the `window.location.href`